packages/salesforcedx-vscode-core/test/jest/commands/retrieveExecutor.test.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import {
+  ConfigUtil,
+  ContinueResponse,
+  SourceTrackingService
+} from '@salesforce/salesforcedx-utils-vscode';
+import { ComponentSet } from '@salesforce/source-deploy-retrieve';
+import * as fs from 'fs';
+import { RetrieveExecutor } from '../../../src/commands/baseDeployRetrieve';
+import { WorkspaceContext } from '../../../src/context/workspaceContext';
+
+jest.mock('../../../src/sfdxProject/sfdxProjectConfig');
+
+jest.mock('../../../src/conflict/metadataCacheService');
+
+describe('Retrieve Executor', () => {
+  const dummyProcessCwd = '/';
+  const dummyComponentSet = new ComponentSet();
+  const mockWorkspaceContext = { getConnection: jest.fn() } as any;
+  const updateTrackingFromRetrieveMock = jest.fn().mockResolvedValue({});
+  const dummySourceTracking = {
+    updateTrackingFromRetrieve: updateTrackingFromRetrieveMock
+  } as any;
+  const dummyRetrieveResult = {
+    components: {},
+    localComponents: {},
+    response: {}
+  } as any;
+  const dummyRetrieveOperation = {
+    pollStatus: jest.fn().mockResolvedValue(dummyRetrieveResult)
+  } as any;
+
+  let workspaceContextGetInstanceSpy: jest.SpyInstance;
```

Comment:
Unused?

Reply:
It's used on line 58 below.  I added an expect into the test for it so that it is being used in the assertions.